ProcessOut/Classes/ProcessOut.swift

```diff
@@ -365,6 +366,34 @@ public class ProcessOut {
         }
     }
     
+    /// Increments the authorization of an applicable invoice by a given amount
+    ///
+    /// - Parameters:
+    ///   - invoiceId: Invoice generated on your backend
+    ///   - amount: The amount by which the authorization should be incremented
+    ///   - handler: Custom 3DS2 handler (please refer to our documentation for this)
+    public static func incrementAuthorizationAmount(invoiceId: String, amount: Int, handler: ThreeDSHandler) {
+        let incrementRequest = IncrementAuthorizationRequest(amount: amount)
+        guard let body = try? JSONEncoder().encode(incrementRequest) else {
+            handler.onError(error: ProcessOutException.InternalError)
```

Comment:
Is there a unit test for if this encoding fails? Would it be better to inject and stub `JSONEncoder`?

Reply:
At the moment, unit tests in this project are quite limited. All the unit tests are contained within this file:

https://github.com/processout/processout-ios/blob/e652924ef47d2404ab6e844eba5409997647202e/Example/ProcessOut_UITests/ProcessOut_UITests.swift

So far fail states have generally not been accounted for in unit tests, including the one for this function. There are plans to improve the structure and coverage of unit tests throughout the project, but that will require changes that are outside the scope of this PR. 

Injecting `JSONEncoder` would be the most logical way to test this fail state in my opinion, but adding an additional argument to the signature of `incrementAuthorizationAmount` (and any other function where JSON encoding takes place) would add complexity to the API without providing much benefit to the end user that I can identify.

I am not aware of any relevant established best practices, so I would be interested to hear your opinion regarding this tradeoff and any potential alternative solutions.